pkg/resources/k8sgpt.go

```diff
@@ -273,6 +274,35 @@ func GetDeployment(config v1alpha1.K8sGPT) (*appsv1.Deployment, error) {
 			},
 		},
 	}
+	if outOfClusterMode {
```

Comment:
shouldn't we standardize the code based on the `Spec` ?

```go
if config.Spec.Kubeconfig {
```

Reply:
> Shouldn't we handle all these changes by updating the version of the k8sgpt resource

It depends on the API governance of the project, although sticking to the best practices of Kubernetes operators development, such as agreed on several Cluster API projects or Kubernetes itself, adding fields to a specification doesn't require a bump of the version since it's a feature addition that would require an update of the deployed Operator.

> This would allow us to maintain two versions of the controller,

In operator development, it's highly discouraged to have two controllers for the same resource, even tho this has multiple API versions. Rather, a version is elected as Hub which is going to be the _stored_ one in etcd, which will be used as "hub" to convert back and forth from other versions.

> preventing any breaking changes and avoiding complicating the migration process

As I stated before, adding fields is not considered as a breaking change. The migration process can be achieved by leveraging the Operator SDK/controller-runtime machinery, such as the conversion webhook which will translate API objects between versions in a transparent mode for the end-user. Adding a conversion requires a lot of more code base, as well as webhooks management (which require CA handling): it should be put in place if and only if definitely required, this doesn't seem the case to me.